pkg/multi-tenancy/write/write.go

```diff
@@ -0,0 +1,39 @@
+// This file and its contents are licensed under the Apache License 2.0.
+// Please see the included NOTICE for copyright information and
+// LICENSE for a copy of the license.
+
+package write
+
+import (
+	"github.com/timescale/promscale/pkg/multi-tenancy/config"
+	"github.com/timescale/promscale/pkg/prompb"
+)
+
+// writeAuthorizer is a write authorizer that authorizes if the incoming write request is valid to be written or not.
+type writeAuthorizer struct {
+	*config.Config
+}
+
+// NewAuthorizer returns a new plainWriteAuthorizer.
+func NewAuthorizer(config *config.Config) Authorizer {
+	return &writeAuthorizer{config}
+}
+
+func (a *writeAuthorizer) IsAuthorized(tenantName string) bool {
+	return a.IsTenantAllowed(tenantName)
+}
+
+func (a *writeAuthorizer) VerifyAndApplyTenantLabel(tenantName string, labels []prompb.Label) ([]prompb.Label, bool) {
+	for _, label := range labels {
+		if label.Name == config.TenantLabelKey {
+			return labels, false
+		}
+	}
+	if tenantName == "" {
+		// Non-multi-tenant write request.
```

Comment:
as I mentioned in the flags section, whether or not to accept data without a tenant name should be a configuration option

Reply:
yup

Reply:
(note to myself) TODO

Reply:
still needs to be done

Reply:
whether or not to allow multi-tenancy to accept non MT write requests is accepted from `multi-tenancy-allow-non-multi-tenant-writes` flag and is checked below this (in the commit).